core/jvm/src/main/scala/cats/effect/IOCompanionPlatform.scala

```diff
@@ -28,6 +28,23 @@ private[effect] abstract class IOCompanionPlatform { this: IO.type =>
   private[this] val TypeInterruptibleOnce = Sync.Type.InterruptibleOnce
   private[this] val TypeInterruptibleMany = Sync.Type.InterruptibleMany
 
+  /**
+   * Intended for thread blocking operations. `blocking` will shift the execution of the
+   * blocking operation to a separate threadpool to avoid blocking on the main execution
+   * context. See the thread-model documentation for more information on why this is necessary.
+   * Note that the created effect will be uncancelable; if you need cancellation then you should
```

Comment:
Minor nit, but we tend to use the alternative spelling, "cancelation".

```suggestion
   * Note that the created effect will be uncancelable; if you need cancelation then you should
```